script.js

```diff
@@ -14,3 +15,127 @@ window.addEventListener("scroll", function () {
 
   prevScrollPos = currentScrollPos;
 });
+
+//START Form validation
+
+const form = document.querySelector("#form");
+const nameEl = document.querySelector("#name");
+const emailEl = document.querySelector("#email");
+const messageEl = document.querySelector("#message");
+
+const validateField = (inputEl, min, max, errorMessage) => {
+  const value = inputEl.value.trim();
+
+  if (!isRequired(value)) {
+    showError(inputEl, errorMessage);
+    return false;
+  }
+
+  if (!isBetween(value.length, min, max)) {
+    showError(inputEl, `Field must be between ${min} and ${max} characters.`);
+    return false;
+  }
+
+  showSuccess(inputEl);
+  return true;
+};
+
+const checkName = () => {
+  return validateField(nameEl, 3, 25, "Name field cannot be blank.");
+};
+
+const checkMessage = () => {
+  return validateField(
+    messageEl,
+    1,
+    Infinity,
+    "Message field cannot be blank."
+  );
+};
+
+const isEmailValid = (email) => {
+  const re =
+    /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+};
+
+const checkEmail = () => {
+  const email = emailEl.value.trim();
+
+  if (!validateField(emailEl, 1, Infinity, "Email field cannot be blank.")) {
+    return false;
+  }
+
+  if (!isEmailValid(email)) {
+    showError(emailEl, "Please enter a valid email address.");
+    return false;
+  }
+
+  showSuccess(emailEl);
+  return true;
+};
+
+const isRequired = (value) => (value === "" ? false : true);
```

Comment:
It would be better to name this one something like `isEmpty` because that is what you are checking for here.